test/spec_request.rb

```diff
@@ -1867,8 +1872,6 @@ def initialize(req)
       end
 
       def delegate?; true; end
-
-      def env; @req.env.dup; end
```

Comment:
@tenderlove since you wrote this code do you mind explaining why you are calling dup here if you recall the reason?

Reply:
IIRC the idea was that if we return the hash directly, then people can muck with it without going through getters / setters.  Looks like this is in a spec though so, I'm not sure why we need it. 😅

Reply:
Yeah, IMHO it was hiding bugs.